lib/validate.js

```diff
@@ -177,7 +177,78 @@ const validateCategory = (
   return errors;
 };
 
+const validateEntitySchema = (
+  ms,
+  { definition: schema },
+  instance,
+  options = {}
+) => {
+  const { path = '', patch = false } = options;
+  const errors = [];
+  const schemaProps = new Set(Object.keys(schema));
+  const objectProps = new Set(Object.keys(instance));
+  const props = new Set([...schemaProps, ...objectProps]);
+  for (const prop of props) {
+    const isSchemaProp = schemaProps.has(prop);
+    const isObjectProp = objectProps.has(prop);
+    if (isObjectProp && !isSchemaProp) {
+      errors.push(new ValidationError('unresolvedProperty', `${path}${prop}`));
```

Comment:
You should also check for `Symbol.for('other')` if the schema has it then you should validate against it.